format/spec.md

```diff
@@ -44,6 +44,15 @@ The primary change in version 2 adds delete files to encode rows that are delete
 
 In addition to row-level deletes, version 2 makes some requirements stricter for writers. The full set of changes are listed in [Appendix E](#version-2).
 
+#### Version 3: Extended Types and Capabilities
```

Comment:
I think there is quite a bit of overlap with https://github.com/apache/iceberg/blob/main/format/spec.md#version-3. Looking at the questions that @emkornfield raised, it feels that this summary is too high level, maybe we should refer to `Appendix E` for details.